felix/bpf/polprog/pol_prog_builder.go

```diff
@@ -317,7 +312,11 @@ func (p *Builder) writeProgramFooter(forXDP bool) {
 	// Execute the tail call to drop program
 	p.b.Mov64(R1, R6)                      // First arg is the context.
 	p.b.LoadMapFD(R2, uint32(p.jumpMapFD)) // Second arg is the map.
-	p.b.MovImm32(R3, jumpIdxDrop)          // Third arg is the index (rather than a pointer to the index).
+	jumpDropIndex := bpf.ProgIndexDrop
+	if p.forIPv6 {
+		jumpDropIndex = bpf.ProgIndexV6Drop
+	}
+	p.b.MovImm32(R3, int32(jumpDropIndex)) // Third arg is the index (rather than a pointer to the index).
```

Comment:
Pehaps hide these ifs in methods?

```suggestion
	p.b.MovImm32(R3, p.jumpDropIndex()) // Third arg is the index (rather than a pointer to the index).
```